consensus/state.go

```diff
@@ -764,15 +764,17 @@ func (cs *ConsensusState) enterPropose(height int, round int) {
 
 	// Nothing more to do if we're not a validator
 	if cs.privValidator == nil {
+		log.Notice("This node is not a validator")
 		return
 	}
 
 	if !bytes.Equal(cs.Validators.GetProposer().Address, cs.privValidator.GetAddress()) {
 		log.Info("enterPropose: Not our turn to propose", "proposer", cs.Validators.GetProposer().Address, "privValidator", cs.privValidator)
+		log.Notice("This node is a validator")
```

Comment:
A node can have a privValidator without it being in the validator set. 

Do the same check as in node/node.go